src/Raven.Client/Documents/Operations/ETL/Elasticsearch/ElasticsearchConnectionString.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Raven.Client.Documents.Operations.ConnectionStrings;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Client.Documents.Operations.ETL.Elasticsearch
+{
+    public class ElasticsearchConnectionString : ConnectionString
+    {
+        public string[] Nodes;
```

Comment:
I need to think about this. I will back to you with the answer.

Reply:
There are few options for authentication.

- ApiKeyAuthentication
- BasicAuthentication
- ClientCertificate
- ClientCertificates

https://www.elastic.co/guide/en/elasticsearch/client/net-api/current/configuration-options.html

Should we support all of them or just some of them? @ayende

Reply:
I'm for ApiKey and Basic auth support

Reply:
I would rather support them all and let the user decide.

Need to ensure that for secured database, we raise a warning if they are sending to an insecured Elastic isntance.

Reply:
1. I wanted to avoid keeping the certificate but OK - we can keep is as base64. 
2. Currently we rise this warning when Elasticsearch instance doesn't have "https" (just "http") in URL - so same logic as with RavenDB. But indeed, we should also warn if there is no authentication required.

Reply:
@djordjedjukic 

Regarding point 2 - I think the best place to check it and warn if needed would be here:

https://github.com/ravendb/ravendb/blob/ab4da6d7416e0174d483f8064d6506a5852ce58c/src/Raven.Server/Documents/ETL/EtlLoader.cs#L333-L334

then you could do

```
if ((_databaseRecord.Encrypted && config is ElasticSearchConfig esConfig && esConfig.Connection.Auth == null)
{
   LogConfigurationWarning()
}

```

Reply:
Note that we don't want to return `false` - the config is still valid but just want to warn a user

Reply:
done